account_move_batch_validate/models/account_move.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2014 Camptocamp SA, 2017 ACSONE
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models, _
+from odoo.addons.queue_job.job import job, Job
+
+
+_logger = logging.getLogger(__name__)
+
+BLOCK_SIZE = 1000
+
+
+class AccountMove(models.Model):
+
+    _inherit = 'account.move'
+
+    to_post = fields.Boolean(
+        string="Posting requested", readonly=True,
+        help="Check this box to mark the move for batch posting")
+    post_job_uuid = fields.Char(string="UUID of the Job to approve this move")
+
+    @job(default_channel='root.account_move_batch_validate')
+    def validate_one_move(self, move_id):
+        move = self.browse(move_id)
+        if move.exists():
+            move.post()
+        else:
+            return _("Nothing to do because the record has been deleted")
+
+    @api.model
+    def _delay_post_marked(self, eta=None):
+        """
+        Create a job for every move marked for posting.
+        If some moves already have a job, they are skipped.
+        """
+        AccountMoveObj = self.env[self._name]
+
+        moves = self.search([
+            ('to_post', '=', True),
+            ('post_job_uuid', '=', False),
+            ('state', '=', 'draft'),
+        ])
+
+        # maybe not creating too many dictionaries will make us a bit faster
+        values = {'post_job_uuid': None}
+        _logger.info(
+            "%s jobs for posting moves have been created.", len(moves))
+
+        for move in moves:
+            new_job = AccountMoveObj.with_delay(eta=eta).validate_one_move(
+                move.id)
+            values['post_job_uuid'] = new_job.uuid
+            move.write(values)
+            self.env.cr.commit()  # pylint:disable=invalid-commit
```

Comment:
@arkostyuk @benwillig any good reason to commit here?
It's because you can have parallel jobs?
If it is really mandatory it would be better to have a comment explaining why :wink:

Reply:
@arkostyuk @benwillig piiiiiiiiiiiiiiiing :smile:

Reply:
Sorry for the delay. I guess we want to be able to track which job has been created for which move during the process, and not only at the end of it. If you want to post a lot of moves at the same time, you can see the job creation in live and you don't have to wait that all moves have been processed. The same is done in ```mark_for_posting``` method.

Reply:
I'm a bit late here, sorry, but IMO user experience is not a good enough reason to put the database at risk.
cf https://www.odoo.com/documentation/11.0/reference/guidelines.html#never-commit-the-transaction

Moreover, looking at the code, there are IMO a few bad design issues that should be better considered. What is happening here, is an UPDATE request is sent for each move to post and is then committed afterwards (on top of the job creation which is also costly, so no surprise this is so slow, the user wants to keep track....). But the only reason seems to be linking the job to the move, but that's a bad idea and the commit is there to solve it... IIRC the move to process is stored on the job anyway, so instead of this, do ONE write on all the moves, and afterwards update all the moves with their job id if this is really needed... 
BTW it's even worse in `mark_for_posting` since `values` don't change between the moves and the `write` method is called inside the for loop.... Just do `moves.write(values)` instead of looping and committing...

Thinking even further, why do we need ONE job PER move ?
A better design would be to check first which moves can't be posted, then exclude them from the recordset and create a single job to process all these moves. There are many possible improvements, but IMO the actual code is bad.